src/librustc_typeck/check/regionck.rs

```diff
@@ -1231,83 +1204,27 @@ impl<'a, 'tcx> RegionCtxt<'a, 'tcx> {
     ///
     /// Here `bk` stands for some borrow-kind (e.g., `mut`, `uniq`, etc).
     ///
-    /// Unfortunately, there are some complications beyond the simple
-    /// scenario I just painted:
-    ///
-    /// 1. The reference `r` might in fact be a "by-ref" upvar. In that
```

Comment:
You probably should transcribe this doc-comment for case (1.) into a doc comment for the currently un-doc-commented `fn link_upvar_region`.